design/sketch.webidl

```diff
@@ -560,7 +567,7 @@ interface WebGPUDevice {
     WebGPUComputePipeline createComputePipeline(WebGPUComputePipelineDescriptor descriptor);
     WebGPURenderPipeline createRenderPipeline(WebGPURenderPipelineDescriptor descriptor);
 
-    WebGPUCommandEncoder createCommandEncoder(WebGPUCommandEncoderDescriptor descriptor);
+    WebGPUCommandBuffer createCommandBuffer(WebGPUCommandBufferDescriptor descriptor);
```

Comment:
For type-safety I think we should consider having this return an encoder, and the encoder return the built commandbuffer. Built command buffers would then be completely opaque objects.

Reply:
what kind of encoder do you want to return from here? I don't think the API complication for making the ready-to-go command buffer type is worth it, since it would mean `WebGPUCommandEncoder -> [WebGPUXxxPassEncoder] -> WebGPUCommandBuffer`. We already have things that you can't do on a command buffer under some circumstances, i.e. you can't do *anything* on it if there is a pass being encoded. Adding a limitation that you can't submit one (while encoding) doesn't seem to be doing anything worse.

Reply:
Sorry I wasn't clear, but you understood what I meant correctly.

It is true that there should be little to no validation overhead, but having a `WebGPUCommandEncoder` return a finalized `WebGPUCommandBuffer` helps in several ways. Essentially the encoder is a mutable reference, while the command buffer is a read-only reference:

 - It is better type-safety
 - This makes it easier to say that unfinalized command buffers cannot be shared across workers but finalized ones can
 - It allows dropping memory used only for validation

Reply:
Alright, the aspect of sharing permissions that may be defined at the WebIDL level is a strong point, thanks for bringing this up! However, I don't think it makes sense if we don't have command buffers re-usability. So how about me just removing the `reusable` flag for now, and then we'll follow up with a PR re-adding it as well as introducing the encoder object in case the group reaches consensus on it?

Reply:
I think I will argue for this split either way b. That said the change you suggested sounds good and we'll discuss more in the group.

Reply:
Even without reusability I strongly think there are benefits to having separate WebGPUCommandEncoder and WebGPUCommandBuffer, especially the worker transferability issue.
```
WebGPUDevice.createCommandEncoder -> WebGPUCommandEncoder (non-transferable)
WebGPUCommandEncoder.beginRenderPass -> WebGPURenderPassEncoder (non-transferable)
WebGPUCommandEncoder.beginComputePass -> WebGPUComputePassEncoder (non-transferable)
WebGPUProgrammablePassEncoder.endPass -> WebGPUCommandEncoder (non-transferable)
WebGPUCommandEncoder.finishEncoding -> WebGPUCommandBuffer (transferable, no leftover validation or encode methods hanging around)
```

Reply:
This is confusing. What's the difference between a WebGPUCommandEncoder and a WebGPURenderPassEncoder?